src/main/java/employeeEnroll/EnrollInteractor.java

```diff
@@ -0,0 +1,44 @@
+package employeeEnroll;
+
+import entity.*;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class EnrollInteractor {
+    final EnrollDsGateway enrolldsGateway;
+    final EnrollPresenter enrollPresenter;
+    final UserFactory userFactory;
+
+
+    public EnrollInteractor(EnrollDsGateway enrolldsGateway, EnrollPresenter enrollPresenter, UserFactory userFactory) {
+        this.enrolldsGateway = enrolldsGateway;
+        this.enrollPresenter = enrollPresenter;
+        this.userFactory = userFactory;
+    }
+
+    public EnrollResponseModel create(EnrollRequestModel requestModel){
+        int id = enrolldsGateway.generateId();
+        String name = requestModel.getName();
+        Department dpt = null; ///////
```

Comment:
Probably we need include department in our request model. Since every employee will assigned to a department when they first come to company

Reply:
Do we assign the department based on department oid?

Reply:
Have changed department as an enum in my pull request.